### PR TITLE
add toBeEvaluated sheet

### PR DESCRIPTION
**Pull request by donnytian**
https://jira.devfactory.com/browse/CODEREVIEW-290162

[//]: # ({"businessId":"","diffLink":"","analyzers":{"js":0,"cs":0,"java":0,"acr":0,"cap":0}})

### CODE REVIEW

**Comment by donnytian** [comment]
Your Pull Request has been selected for a project-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members).

<details><summary>More information</summary>

_This is an initiative to improve overall code quality. If your change is **urgent**, you **don't** have to wait for our review._ 

_NOTE: This review **does not** replace your team's review._

</details>

File: CapReport/GoogleSheetClient.cs [hunk 1]
```diff
@@ -85,29 +86,50 @@ public GoogleSheetClient(dynamic settings, ReportMode mode)
...
+            };
+            var batchRequest = new BatchUpdateSpreadsheetRequest
+            {
+                Requests = new List<Request> { new Request { AppendCells = appendCellsRequest } }
+            };
+            var request = service.Spreadsheets.BatchUpdate(batchRequest, _spreadsheetId);
+            request.Execute();
+        }
+
+        public Sheet FindOrCreateSheet(SheetsService service, string sheetName)
```
Comment (donnytian) [inline CapReport/GoogleSheetClient.cs]: :warning: **Visibility Can Be Tightened**  
  
This member or type scope (Access Modifiers) is wider than needed. To achieve the best encapsulation (information hiding) we should always declare such artifacts with the least visibility that works. By reducing visibility, we reduce coupling.  
  
As a rule of thumb, starting out with `private` and then relaxing the level down to what is needed, will help avoid any encapsulation issues most of the time.

**[More details](https://confluence.devfactory.com/x/I41oGg)**

## The improvement suggestion in this case

Tightening the visibility to **private**.

[//]: # ({"businessId":"efa24ee3-f65a-4440-b4c3-c04b65bbea46","diffLink":"diff-6d0685a18dab77d0cbcfcc5df32e5f08R120","analyzers":{"js":2,"cs":2,"java":2,"acr":2,"cap":2}}) 

<a id="65ae2c50-6348-11ea-a5c5-2b93756ce7af" href="none" codereview-uuid-reserved-tag />

File: CapReport/GoogleSheetClient.cs [hunk 2]
```diff
@@ -219,7 +241,11 @@ private static void AppendHeaderRow(List<RowData> rows)
             rows.Add(new RowData { Values = new List<CellData>(headers) });
         }
 
-        private void AppendDataRows(List<RowData> rows, IEnumerable<CapResult> entries, Finding[] findings)
+        private void AppendDataRows(
```
Comment (donnytian) [inline CapReport/GoogleSheetClient.cs]: :exclamation: **This method implementation is too long.**  
  
A possible improvement would be to extract the newly added logic into a new method or split up the whole implementation into smaller methods.  
  
This change would avoid increasing the present method complexity and improve readability and maintainability, by making the code easier to understand. 

**[More details](https://confluence.devfactory.com/x/hIFkF)**
 
## The solution in this case

*Proof of Concept*   
Consider extracting parts of the code as smaller and private methods.  
Visual Studio has [refactor tools to extract methods](https://docs.microsoft.com/en-us/visualstudio/ide/reference/extract-method) and [Preview Changes window](https://docs.microsoft.com/en-us/visualstudio/ide/preview-changes) to visualize how the refactor will look like.  
When defining the new methods, consider using names that reveal the intention of each extracted code. Sometimes, when doing this enhancement, there are opportunities to also do some extra refactor, consider checking [Microsoft documentation](https://docs.microsoft.com/en-us/visualstudio/ide/refactoring-in-visual-studio) about some improvements that might be applicable to this situation.  
Repeat the operation until the refactored code has methods no longer than 80 lines.  
> NOTE:  
>- Consider applying the structure of this solution in other parts of the code, if there is another occurrence(s).

[//]: # ({"businessId":"efa24ee3-f65a-4440-b4c3-c04b65bbea46","diffLink":"diff-6d0685a18dab77d0cbcfcc5df32e5f08R244","analyzers":{"js":2,"cs":1,"java":2,"acr":2,"cap":2}}) 

<a id="6bc0e420-6348-11ea-a5c5-2b93756ce7af" href="none" codereview-uuid-reserved-tag />

**Comment by donnytian** [comment]
Hi, I'm Donny Tian from [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members) and I have just reviewed your PR.
<b>How likely are you to recommend the Code Review service to another developer?</b>

Not at all likely \[ [1](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/coderead-cap-report/pull/6&entry.1163746017=1) \] \[ [2](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/coderead-cap-report/pull/6&entry.1163746017=2) \] \[ [3](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/coderead-cap-report/pull/6&entry.1163746017=3) \] \[ [4](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/coderead-cap-report/pull/6&entry.1163746017=4) \] \[ [5](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/coderead-cap-report/pull/6&entry.1163746017=5) \] \[ [6](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/coderead-cap-report/pull/6&entry.1163746017=6) \] \[ [7](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/coderead-cap-report/pull/6&entry.1163746017=7) \] \[ [8](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/coderead-cap-report/pull/6&entry.1163746017=8) \] \[ [9](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/coderead-cap-report/pull/6&entry.1163746017=9) \] \[ [10](https://docs.google.com/a/aurea.com/forms/d/e/1FAIpQLSdEMTCUcOnA4z86LUOWodIg0M6XQpQT83QtLRg_xgjeHOPPoA/viewform?entry.113507233=https://github.com/trilogy-group/coderead-cap-report/pull/6&entry.1163746017=10) \] Very likely

> TIP: Press `Ctrl` (or `⌘` in a Mac) to open the survey in a new tab.

**Comment by amit-chaurasia** [comment]
Your Pull Request has been selected for a project-agnostic code review by [Aurea Central Reviewers Team](https://github.com/orgs/trilogy-group/teams/aurea-central-reviewers/members).

<details><summary>More information</summary>

_This is an initiative to improve overall code quality. If your change is **urgent**, you **don't** have to wait for our review._ 

_NOTE: This review **does not** replace your team's review._

</details>

[//]: # ({"businessId":"","diffLink":"","analyzers":{"js":0,"cs":0,"java":0,"acr":0,"cap":0}})

**Comment by amit-chaurasia** [comment]
:white_check_mark: No issues identified in this pull request. 
    
**Good work!**
<details>
    <summary>T4W FTAR Statistics</summary>
    <br />
    CodeRead team introduced new quality measurement to evaluate the trailing 4 weeks First Time Acceptance Rate of Pull Requests, this PR increased your FTAR from 88.89 % to 89.47 %, good job, and for more quality improvements you can read about yellow findings.
	<br />
    -
    <details>
        <summary>More Details</summary>
        <br />
        <table>
            <tr>
                <td>Current FTAR</td>
                <td><b>89.47 %</b></td>
            </tr>
            <tr>
                <td>Total Reviewed Pull Requests for the last 4 weeks</td>
                <td>19</td>
            </tr>
            <tr>
                <td>Passed</td>
                <td>17</td>
            </tr>
            <tr>
                <td>Rejected</td>
                <td>2</td>
            </tr>
        </table>
        More details about how FTAR is calculated, refer to this page - <a
            href="https://confluence.devfactory.com/x/Cw8dGg">FTAR - First time acceptance rate</a>
    </details>

</details>

**Pull request merged**